src/appkey.rs

```diff
@@ -10,48 +10,63 @@ fn read_file_contents(path: &str) -> Result<String, io::Error> {
     Ok(content)
 }
 
+fn get_appkey_from_keyring() -> Option<String> {
+    let output = Command::new("secret-tool")
+        .arg("lookup")
+        .arg("htb-api")
+        .arg("user-htb-api")
+        .output()
+        .ok()?;
+    
+    if output.status.success() {
+        let key = String::from_utf8_lossy(&output.stdout)
+            .trim()
+            .to_string();
+        if !key.is_empty() {
+            return Some(key);
+        }
+    }
+    None
+}
+
 pub fn get_appkey() -> String {
+    // Try Docker secret first
     if is_inside_container() && is_display_empty() {
         let secret_path = "/run/secrets/htb-api";
-        
+
         match read_file_contents(secret_path) {
             Ok(secret_content) => {
-                
-                secret_content.replace('\n', "")
+                let key = secret_content.replace('\n', "");
+                if !key.is_empty() {
+                    return key;
+                }
             }
             Err(error) => {
-                if error.kind() == io::ErrorKind::NotFound {
-                    eprintln!("File not found");
-                    String::new()
-                } else {
-                    eprintln!("Error: {error}");
-                    String::new()
+                if error.kind() != io::ErrorKind::NotFound {
+                    eprintln!("Error reading Docker secret: {error}");
                 }
             }
         }
-    } else {
-        let output = Command::new("secret-tool")
-            .arg("lookup")
-            .arg("htb-api")
-            .arg("user-htb-api")
-            .output();
-
-        match output {
-            Ok(output) => {
-                if output.status.success() {
-                    String::from_utf8_lossy(&output.stdout).to_string()
-                } else {
-                    let error_output = String::from_utf8_lossy(&output.stderr);
-                    eprintln!("'secret-tool' command failed:\n{error_output}");
-                    String::new()
-                }
-            }
-            Err(error) => {
-                eprintln!("Error: {error}");
-                String::new()
-            }
+    }
+
+    // Try keyring
+    if let Some(key) = get_appkey_from_keyring() {
+        return key;
+    }
+
+    // Fallback to config file
+    let config_path = format!("{}/.config/htb-toolkit/token", 
```

Comment:
```suggestion
    let config_path = format!("{}/.config/htb-toolkit/token", 
```
I understand your point... even if it makes me worried. I hope that people can follow the `chmod 600` you added on README.